lib/datadog/statsd.rb

```diff
@@ -337,35 +343,69 @@ def format_event(title, text, opts={})
     private_constant :NEW_LINE, :ESC_NEW_LINE, :COMMA, :BLANK, :PIPE, :DOT,
       :DOUBLE_COLON, :UNDERSCORE
 
-    def escape_event_content(msg)
-      msg.gsub NEW_LINE, ESC_NEW_LINE
+    def escape_event_content!(msg)
+      msg.gsub!(NEW_LINE, ESC_NEW_LINE) || msg
     end
 
-    def escape_tag_content(tag)
-      remove_pipes(tag).gsub COMMA, BLANK
+    def escape_tag_content!(tag)
```

Comment:
I think I addressed everything in https://github.com/DataDog/dogstatsd-ruby/pull/40/commits/b2d1ca6cc07bad93cc41ccdaeeb691550db7a01d

I did another benchmark after adding the `dup`s and it's still an improvement over what we had:

```
Calculating -------------------------------------
              master   128.800M memsize (     0.000  retained)
                         2.300M objects (     0.000  retained)
                        50.000  strings (     0.000  retained)
      current branch    79.900M memsize (     0.000  retained)
                         1.400M objects (     0.000  retained)
                        50.000  strings (     0.000  retained)

Comparison:
      current branch:   79900050 allocated
              master:  128800050 allocated - 1.61x more
```

Can you both give this another look? 🙏 💖